lib/foreignNodeId.ts

```diff
@@ -0,0 +1,65 @@
+import { DirectiveLocation, GraphQLNonNull, GraphQLString } from 'graphql';
+import { ValidationError } from 'apollo-server-errors';
+
+import ValidateDirectiveVisitor, {
+  ValidateFunction,
+} from './ValidateDirectiveVisitor';
+
+export type ToNodeId = (id: string) => { typename: string; id: string } | null;
```

Comment:
very minor, but I wonder if it should be a generic on the type of `id`, since in some cases it will be a number or the likes, ex:

```typescript
export type ToNodeId<IdType> = (id: string) => { typename: string; id: IdType } | null;
export type ForeignNodeIdContext<IdType, TContext extends object = object> = {
  fromNodeId: ToNodeId<IdType>;
};
```